code/game/objects/items/robot/robot_parts.dm

```diff
@@ -62,6 +62,54 @@
 				return 1
 	return 0
 
+/obj/item/robot_suit/wrench_act(mob/living/user, obj/item/I) //Deconstucts empty borg shell. Flashes remain unbroken because they haven't been used yet
+	var/turf/T = get_turf(src)
+	forceMove(T)
+	if(l_leg || r_leg || chest || l_arm || r_arm || head)
+		if(I.use_tool(src, user, 5, volume=50))
+			if(l_leg)
+				l_leg.forceMove(T)
+				l_leg = null
+			if(r_leg)
+				r_leg.forceMove(T)
+				r_leg = null
+			if(chest)
+				if (chest.cell) //Sanity check.
+					chest.cell.forceMove(T)
+					chest.cell = null
+				chest.forceMove(T)
+				new /obj/item/stack/cable_coil(T, 1)
+				chest.wired = 0
+				chest = null
+			if(l_arm)
+				l_arm.forceMove(T)
+				l_arm = null
+			if(r_arm)
+				r_arm.forceMove(T)
+				r_arm = null
+			if(head)
+				head.forceMove(T)
+				head.flash1.forceMove(T)
+				head.flash1 = null
+				head.flash2.forceMove(T)
+				head.flash2 = null
+				head = null
+			to_chat(user, "<span class='notice'>You disassemble the cyborg shell.</span>")
+	else
+		to_chat(user, "<span class='notice'>There is nothing to remove from the endoskeleton.</span>")
+	updateicon()
+
+obj/item/robot_suit/screwdriver_act(mob/living/user, obj/item/I) //Swaps the power cell if you're holding a new one in your other hand.
+	var/turf/T = get_turf(src)
+	if(istype(user.held_items[user.active_hand_index == 1 ? 2 : 1], /obj/item/stock_parts/cell))
+		if (chest.cell) //Sanity check.
+			chest.cell.forceMove(T)
+		chest.cell = user.held_items[user.active_hand_index == 1 ? 2 : 1]
+		if(!user.transferItemToLoc(user.held_items[user.active_hand_index == 1 ? 2 : 1], src.chest))
+			chest.cell = null
+			return
+		to_chat(user, "<span class='notice'>You replace the power cell.</span>")

```

Comment:
```
/obj/item/robot_suit/screwdriver_act(mob/living/user, obj/item/I) //Swaps the power cell if you're holding a new one in your other hand.
	var/obj/item/stock_parts/cell/temp_cell = user.is_holding_item_of_type(/obj/item/stock_parts/cell)
	if(!temp_cell)
		return
	if(!user.transferItemToLoc(temp_cell, chest))
		to_chat(user, "<span class='warning'>[temp_cell] is stuck to your hand, you can't put it in [src]!</span>")
		return
	if(!QDELETED(chest.cell))
		if(!user.put_in_hands(chest.cell))
			chest.cell.forceMove(drop_location())
	chest.cell = temp_cell
	to_chat(user, "<span class='notice'>You replace [src]'s power cell.</span>")
	return TRUE
```
this will put the old cell in the user's hands instead.

Reply:
the line `to_chat(user, "<span class='notice'>There doesn't seem to unscrew from [src].</span>")` is for if no cell is found inside the shell, right?